UnityProject/Assets/Scripts/Learning/ProtipManager.cs

```diff
@@ -39,13 +45,53 @@ public override void Awake()
 		private void OnEnable()
 		{
 			UpdateManager.Add(CheckQueue, 1f);
+			UpdateRecordedTips();
 		}
 
 		private void OnDisable()
 		{
 			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, CheckQueue);
 		}
 
+		public void SaveTipState(string ID, bool dontShowAgain = true)
+		{
+			UpdateRecordedTips();
+			var found = false;
+			foreach (var tip in ProtipSaveStates)
+			{
+				if (tip.ID != ID) continue;
+				tip.Remembered = dontShowAgain;
+				found = true;
+				break;
+			}
+			if(found == false)
+			{
+				var newTip = new ProTipSaveData { ID = ID, Remembered = dontShowAgain };
+				ProtipSaveStates.Add(newTip);
+			}
+
+			SaveProtipSaveStates();
+		}
+
+		private void UpdateRecordedTips()
+		{
+			if (File.Exists(jsonPath) == false || File.ReadAllText(jsonPath).Length <= JSON_EMPTY_LIST) return;
+			var newList = JsonConvert.DeserializeObject<List<ProTipSaveData>>(File.ReadAllText(jsonPath));
+			ProtipSaveStates = newList;
```

Comment:
As this comes from IO we can't guarantee the string is always deserializable.  `DeserializeObject()` might return null or throw (check documentation).

At the least, null check it:
```suggestion
			var newList = JsonConvert.DeserializeObject<List<ProTipSaveData>>(File.ReadAllText(jsonPath));
			ProtipSaveStates = newList ?? new List<ProTipSaveData>();
```

Reply:
this causes an NRE when Rider suggested it

Reply:
handle it wherever you use ``ProtipSaveStates``.